File: app/src/main/java/com/alphawallet/app/viewmodel/AdvancedSettingsViewModel.java
package com.alphawallet.app.viewmodel;

import android.os.Environment;

import com.alphawallet.app.repository.PreferenceRepositoryType;
import com.alphawallet.app.service.TransactionsService;

import java.io.File;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;
import io.reactivex.Single;

@HiltViewModel
public class AdvancedSettingsViewModel extends BaseViewModel {
    private final PreferenceRepositoryType preferenceRepository;
    private final TransactionsService transactionsService;

    @Inject
    AdvancedSettingsViewModel(
            PreferenceRepositoryType preferenceRepository,
            TransactionsService transactionsService) {
        this.preferenceRepository = preferenceRepository;
        this.transactionsService = transactionsService;
    }

    public boolean createDirectory() {
        //create XML repository directory
        File directory = new File(
                Environment.getExternalStorageDirectory()
                        + File.separator + HomeViewModel.ALPHAWALLET_DIR);

        if (!directory.exists()) {
            return directory.mkdir();
        }
        else
        {
            return directory.exists();
        }
    }

    public void setFullScreenState(boolean state)
    {
        preferenceRepository.fullScreenState = state;
    }

    public void toggle1559Transactions(boolean toggleState)
    {
        preferenceRepository.use1559Transactions = toggleState;
    }

    public boolean get1559TransactionsState()
    {
        return preferenceRepository.use1559Transactions;
    }

    public boolean getDeveloperOverrideState()
    {
        return preferenceRepository.developerOverride;
    }

    public void toggleDeveloperOverride(boolean toggleState)
    {
        preferenceRepository.developerOverride = toggleState;
    }

    public boolean getFullScreenState()
    {
        return preferenceRepository.fullScreenState;
    }

    public void blankFilterSettings()
    {
        preferenceRepository.blankHasSetNetworkFilters();
    }

    public Single<Boolean> resetTokenData()
    {
        return transactionsService.wipeDataForWallet();
    }

    public void stopChainActivity()
    {
        transactionsService.stopActivity();
    }

    public void toggleUseViewer(boolean state)
    {
        preferenceRepository.useTSViewer = state;
    }

    public boolean getTokenScriptViewerState()
    {
        return preferenceRepository.useTSViewer;
    }
}

File: app/src/main/java/com/alphawallet/app/viewmodel/AdvancedSettingsViewModel.kt
package com.alphawallet.app.viewmodel

import android.os.Environment
import com.alphawallet.app.repository.PreferenceRepositoryType
import com.alphawallet.app.service.TransactionsService
import dagger.hilt.android.lifecycle.HiltViewModel
import java.io.File
import javax.inject.Inject

@HiltViewModel
class AdvancedSettingsViewModel @Inject internal constructor(
    private val preferenceRepository: PreferenceRepositoryType,
    private val transactionsService: TransactionsService
) : BaseViewModel() {
    fun createDirectory(): Boolean {
        //create XML repository directory
        val directory = File(
            (Environment.getExternalStorageDirectory()
                .toString() + File.separator + HomeViewModel.ALPHAWALLET_DIR)
        )

        if (!directory.exists()) {
            return directory.mkdir()
        } else {
            return directory.exists()
        }
    }

    fun toggle1559Transactions(toggleState: Boolean) {
        preferenceRepository.use1559Transactions = toggleState
    }

    val transactions1559State : Boolean
        get() = preferenceRepository.use1559Transactions

    val developerOverrideState: Boolean
        get() = preferenceRepository.developerOverride

    fun toggleDeveloperOverride(toggleState: Boolean) {
        preferenceRepository.developerOverride = toggleState
    }

    var fullScreenState: Boolean
        get() = preferenceRepository.fullScreenState
        set(state) {
            preferenceRepository.fullScreenState = state
        }

    fun blankFilterSettings() {
        preferenceRepository.blankHasSetNetworkFilters()
    }

    suspend fun resetTokenData(): Boolean{
        return transactionsService.wipeDataForWalletSuspend()
    }

    fun stopChainActivity() {
        transactionsService.stopActivity()
    }

    fun toggleUseViewer(state: Boolean) {
        preferenceRepository.useTSViewer = state
    }

    val tokenScriptViewerState: Boolean
        get() = preferenceRepository.useTSViewer
}
